cmd/zdb/zdb.c

```diff
@@ -8246,8 +8246,8 @@ main(int argc, char **argv)
 			dump_opt[c] += verbose;
 	}
 
-	aok = (dump_opt['A'] == 1) || (dump_opt['A'] > 2);
-	zfs_recover = (dump_opt['A'] > 1);
+	aok = (dump_opt['A'] == 1) || (dump_opt['A'] + verbose > 2);
+	zfs_recover = (dump_opt['A'] - verbose > 1);
```

Comment:
I understand the change to `zfs_recover` but not to `aok`.  Can you explain how this works?

Alternatively, could we choose to make `verbose` not bump `dump_opt['A']`?  (Similar to how `dump_all` doesn't set `dump_opt['A']`)

Reply:
What I saw was that `dump_opt['A']` is initially set to 1 when -A is passed. `verbose` always defaults to `1` even when there's no `-v` present so `dump_opt['A']` is always sitting at 2 by this point. Since the check was looking for `== 1` or `> 2` it would never work. I added the `verbose` value to `dump_opt['A']` so that these kinds of scenarios would work:
```
zdb -A
dump_opt['A'] -> initially set to 1
dump_opt['A'] += verbose is always defaulted to 1 so dump_opt['A'] becomes 2 now
(dump_opt['A'] == 1) || (dump_opt['A'] + verbose > 2); == (2 == 1) || (2 + 1 > 2) -> The last check now passes
```
```
zdb -AA
dump_opt['A'] -> initially set to 2
dump_opt['A'] += verbose is always defaulted to 1 so dump_opt['A'] becomes 3 now
(dump_opt['A'] == 1) || (dump_opt['A'] + verbose > 2); == (3 == 1) || (3 + 1 > 2) -> The last check now passes
```
```
zdb -A -vv
dump_opt['A'] -> initially set to 1
dump_opt['A'] += verbose is 2 so dump_opt['A'] becomes 3 now
(dump_opt['A'] == 1) || (dump_opt['A'] + verbose > 2); == (3 == 1) || (3 + 2 > 2) -> The last check now passes
```
etc.

I really like the idea of making `verbose` not affect the opts. I think that will make the fix more straightforward. I'm pretty new to the ZFS code base (really wanting to learn more) so I'll have to look through this whole area and make sure I don't screw something up as part of decoupling `verbose`. Will take a stab at it.

Reply:
I checked the zdb manpage and found:
```
     -A      Do not abort should any assertion fail. [i.e. set `aok`]

     -AA     Enable panic recovery, certain errors which would otherwise be fatal are demoted to warnings. [i.e. set `zfs_recover`]

     -AAA    Do not abort if asserts fail and also enable panic recovery. [i.e. set both]
```

So I think that `-AA` should set `zfs_recover` but not set `aok`.  It sounds like that isn't happening in your example.

Thanks for taking a look at this!  Hopefully the idea of decoupling `verbose` works out.